src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java

```diff
@@ -161,7 +161,7 @@ public Transaction getTransactionSubmission() {
         }
 
         Transaction t = new Transaction(date, type, amount, account,
-                payee, pending, category, notes);
+                payee, pending, category, notes, Integer.parseInt(checkNo));
```

Comment:
So This works when there is a check but it doesnt if a different type is selected. You get a invalid number exception. Probably an else statement above setting it to -1 or something would work.